### PR TITLE
Dev/board

### PR DESCRIPTION
**Pull request by arjund-06**
### Functionality:
Allows the user logged in to see all his tasks right from the first sprint.

### Solution:
Created a task board button and sent the user name and filtered tasks that had the user's name as assignee. 

### Risk level:
- [ ] high 
- [X] medium
- [ ] low

### How to test:
ng serve

## How it looks when user has logged in:

When a user who is logged in opens worktrolly he shall see the my DashBoard page first.

### taskBoard page -> myDashBoard

image

### Dashboard -> Board

image

## How it looks when user is not logged in:

On opening worktrolly without being logged in will redirect to board instead of My DashBoard.

image

### CODE REVIEW

**Comment by viv9k** [comment]
It would be better if we have this component as Dashboard, and move current component as board.

Also make this list box a little small.

**Review by shivamgoyal7 — CHANGES_REQUESTED** [review]
remove the back button from the front page and add the back button in the dashboard.
Also, hide the dashboard button when we are in the dashboard page

File: worktrolly-ui/src/app/body/task-board/task-board.component.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  displayName: string
+  
+  tasksCollection: AngularFirestoreCollection<Tasks>
+  tasksData: Observable<TasksId[]>
+  userObservable: Observable<User>
+  // constructor(public authService: AuthService, public router: Router) { }
+
+  
+  
+
```
Comment (shivamgoyal7) [inline worktrolly-ui/src/app/body/task-board/task-board.component.ts:26]: remove all the extra space

File: worktrolly-ui/src/app/body/task-board/task-board.component.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  constructor(public router: Router, private db: AngularFirestore,public authService: AuthService) { }
+
+  ngOnInit(): void {   
+    
+    this.readUser();
+    
+    
+  }
+
```
Comment (shivamgoyal7) [inline worktrolly-ui/src/app/body/task-board/task-board.component.ts:35]: remove all the extra space

File: worktrolly-ui/src/app/body/task-board/task-board.component.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    this.userObservable = this.authService.afauth.user.pipe(map(action => {
+      const data = action as User;
+      this.user = data;
+      this.displayName = data.displayName;
+      this.readTaskData(this.displayName);
+      return { ...data }
+    }));
+    
+  }
+
```
Comment (shivamgoyal7) [inline worktrolly-ui/src/app/body/task-board/task-board.component.ts:46]: remove all the extra space

**Review by shivamgoyal7 — COMMENTED** [review]
remove extra space from everywhere.
Add back button in the dashboard.
my dashboard(board) page is visible when the user is logged in, when the user is logged off then only the dashboard button is visible so anyone can go on the board.

**Review by viv9k — APPROVED** [review]
Ship it

**Pull request merged**